libraries/UHS_host/UHS_ADK/UHS_ADK.h

```diff
@@ -0,0 +1,133 @@
+/* Copyright (C) 2011 Circuits At Home, LTD. All rights reserved.
+
+This software may be distributed and modified under the terms of the GNU
+General Public License version 2 (GPL2) as published by the Free Software
+Foundation and appearing in the file GPL2.TXT included in the packaging of
+this file. Please note that GPL2 Section 2[b] requires that all works based
+on this software must also be made publicly available under the terms of
+the GPL2 ("Copyleft").
+
+Contact information
+-------------------
+
+Circuits At Home, LTD
+Web      :  http://www.circuitsathome.com
+e-mail   :  support@circuitsathome.com
+ */
+
+/* Google ADK interface support header */
+
+#if !defined(__UHS_ADK_H__)
+#define __UHS_ADK_H__
+
+#include <UHS_host.h>
+
+#define ADK_VID   0x18D1
+#define ADK_PID   0x2D00
+#define ADB_PID   0x2D01
+
+#define XOOM  //enables repeating getProto() and getConf() attempts
+//necessary for slow devices such as Motorola XOOM
+//defined by default, can be commented out to save memory
+
+/* requests */
+
+#define ADK_GETPROTO      51  // check USB accessory protocol version
+#define ADK_SENDSTR       52  // send identifying string
+#define ADK_ACCSTART      53  // start device in accessory mode
+
+#define bmREQ_ADK_GET     USB_SETUP_DEVICE_TO_HOST|USB_SETUP_TYPE_VENDOR|USB_SETUP_RECIPIENT_DEVICE
+#define bmREQ_ADK_SEND    USB_SETUP_HOST_TO_DEVICE|USB_SETUP_TYPE_VENDOR|USB_SETUP_RECIPIENT_DEVICE
+
+#define ACCESSORY_STRING_MANUFACTURER   0
+#define ACCESSORY_STRING_MODEL          1
+#define ACCESSORY_STRING_DESCRIPTION    2
+#define ACCESSORY_STRING_VERSION        3
+#define ACCESSORY_STRING_URI            4
+#define ACCESSORY_STRING_SERIAL         5
+
+#define ADK_MAX_ENDPOINTS 3 // endpoint 0, bulk_IN, bulk_OUT
+
+#if DEBUG_PRINTF_EXTRA_HUGE
+#ifdef DEBUG_PRINTF_EXTRA_HUGE_ADK_HOST
+#define ADK_HOST_DEBUG(...) printf(__VA_ARGS__)
+#else
+#define ADK_HOST_DEBUG(...) VOID0
+#endif
+#else
+#define ADK_HOST_DEBUG(...) VOID0
+#endif
+
+class ADK : public UHS_USBInterface {
+protected:
```

Comment:
Don't protect or make private anything on base classes, as this makes it more difficult to subclass.